### PR TITLE
Build bundles

### PR DESCRIPTION
**Pull request by ktsakalozos**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -97,6 +97,64 @@ add a tag to your github repository.
...
+Note that you can choose to upgrade only a subset of charms. Furthermore,
+for each charm you can specify the channel in which the CI should look for new revisions
+and (optionally) channel the charm should be released to in case of a successful
+test.
+
+An example run of this action might look like this:
+
+    juju run-action cwr/0 build-bundle \
+      repo=https://github.com/juju-solutions/bunlde-cwr-ci \
+      branch=build-bunlde  \
```
Comment (petevg) [inline README.md]: typo

File: README.md [hunk 2]
```diff
@@ -97,6 +97,64 @@ add a tag to your github repository.
 Combining the two jobs gives you a basic yet powerful CI workflow.
 
 
+# Using CWR to CI your Bunldes
+
+## Build Bunlde action
```
Comment (petevg) [inline README.md]: typo

**Comment by petevg** [comment]
I ran this on my local cluster. bundletester griped about timeouts, but otherwise seemed to execute. I am +1 on merging after fixing typos in the docs.

**Comment by petevg** [comment]
Just FYI, here's the console output from my runs:

build_bundle_cwr_ci:

```
Started by upstream project "check-bundle-cwr-ci" build number 14
originally caused by:
 Started by timer
Building in workspace /var/lib/jenkins/jobs/build-bundle-cwr-ci/workspace
[workspace] $ /bin/bash /tmp/hudson2423647870823463001.sh
raven (controller) (no change)
Added 'test-cwr-ci-2' model on localhost/localhost for user 'ravencibuildbundles'
Running git clone https://github.com/juju-solutions/bundle-cwr-ci --branch bundle-building /tmp/tmpz16lin1k/
Cloning into '/tmp/tmpz16lin1k'...
Checking https://github.com/juju-solutions/bundle-cwr-ci
Charms in bundle ['cs:~kos.tsakalozos/jenkins', 'cs:~juju-solutions/cwr']
jenkins
Running charm show cs:~kos.tsakalozos/jenkins -c stable id
id:
  Id: cs:~kos.tsakalozos/jenkins-1
  Name: jenkins
  Revision: 1
  User: kos.tsakalozos
Latest cs:~kos.tsakalozos/jenkins-1
cwr
Running charm show cs:~juju-solutions/cwr -c stable id
id:
  Id: cs:~juju-solutions/cwr-22
  Name: cwr
  Revision: 22
  User: juju-solutions
Latest cs:~juju-solutions/cwr-22
Upgrading cs:~juju-solutions/cwr
Upgrading info {'from-channel': 'edge', 'to-channel': 'beta', 'release': True}
Running charm show cs:~juju-solutions/cwr -c edge id
id:
  Id: cs:~juju-solutions/cwr-26
  Name: cwr
  Revision: 26
  User: juju-solutions
Upgrading to revision cs:~juju-solutions/cwr-26 of channel edge
Running charm show cs:~juju-solutions/cwr -c edge id
id:
  Id: cs:~juju-solutions/cwr-26
  Name: cwr
  Revision: 26
  User: juju-solutions
Testing new bundle
Running cwr -F raven:test-cwr-ci-2 totest.yaml --results-dir /srv/artifacts --test-id 2
2017-01-14 02:30:11 INFO Running test on lxd.
The directory '/var/lib/jenkins/.cache/pip/http' or its parent directory is not owned by the current user and the cache has been disabled. Please check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
The directory '/var/lib/jenkins/.cache/pip' or its parent directory is not owned by the current user and caching wheels has been disabled. check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
Requirement already up-to-date: amulet in /usr/local/lib/python3.5/dist-packages
Requirement already up-to-date: requests in /usr/local/lib/python3.5/dist-packages
Requirement already up-to-date: retrying in /usr/local/lib/python3.5/dist-packages
Requirement already up-to-date: python-jenkins in /usr/local/lib/python3.5/dist-packages
Requirement already up-to-date: libcharmstore in /usr/local/lib/python3.5/dist-packages (from amulet)
Requirement already up-to-date: PyYAML in /usr/local/lib/python3.5/dist-packages (from amulet)
Requirement already up-to-date: path.py in /usr/local/lib/python3.5/dist-packages (from amulet)
Requirement already up-to-date: six>=1.7.0 in /usr/local/lib/python3.5/dist-packages (from retrying)
Requirement already up-to-date: pbr<2.0,>=0.8.2 in /usr/local/lib/python3.5/dist-packages (from python-jenkins)
Requirement already up-to-date: multi-key-dict in /usr/local/lib/python3.5/dist-packages (from python-jenkins)
Requirement already up-to-date: theblues<1.0,>=0.3.7 in /usr/local/lib/python3.5/dist-packages (from libcharmstore->amulet)
Requirement already up-to-date: jujubundlelib>=0.4.1 in /usr/local/lib/python3.5/dist-packages (from theblues<1.0,>=0.3.7->libcharmstore->amulet)
You are using pip version 8.1.2, however version 9.0.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
2017-01-14 02:37:19 ERROR Timeout exceeded. Failed to destroy all applications  in 60 seconds.
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/bundletester/runner.py", line 183, in _run_test
    self.builder.reset()
  File "/usr/local/lib/python2.7/dist-packages/bundletester/builder.py", line 140, in reset
    ' in %s seconds.' % timeout)
RuntimeError: Timeout exceeded. Failed to destroy all applications  in 60 seconds.
2017-01-14 02:38:19 ERROR Timeout exceeded. Failed to destroy all applications  in 60 seconds.
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/bundletester/runner.py", line 183, in _run_test
    self.builder.reset()
  File "/usr/local/lib/python2.7/dist-packages/bundletester/builder.py", line 140, in reset
    ' in %s seconds.' % timeout)
RuntimeError: Timeout exceeded. Failed to destroy all applications  in 60 seconds.
2017-01-14 02:56:37 INFO   Waiting for machine termination
2017-01-14 03:01:47 INFO   Waiting for machine termination
2017-01-14 03:06:48 INFO   Waiting for machine termination
2017-01-14 03:11:53 INFO   Waiting for machine termination
2017-01-14 03:23:39 INFO Starting new HTTP connection (1): svg.juju.solutions
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-cwr-0/charm/scripts/bundlebuilder.py", line 454, in <module>
    tester.test_and_release_bundle(repo, branch, build_num, models)
  File "/var/lib/juju/agents/unit-cwr-0/charm/scripts/bundlebuilder.py", line 419, in test_and_release_bundle
    bundle.test(build_num, models)
  File "/var/lib/juju/agents/unit-cwr-0/charm/scripts/bundlebuilder.py", line 232, in test
    execute(cmd)
  File "/var/lib/juju/agents/unit-cwr-0/charm/scripts/bundlebuilder.py", line 37, in execute
    raise Exception("Command {} failed with {}".format(" ".join(cmd), p.returncode))
Exception: Command cwr -F raven:test-cwr-ci-2 totest.yaml --results-dir /srv/artifacts --test-id 2 failed with 1
Destroying model
Waiting on model to be removed, 1 machine(s)...
Waiting on model to be removed, 1 machine(s)...
Waiting on model to be removed, 1 machine(s)...
Waiting on model to be removed, 1 machine(s)...
Waiting on model to be removed, 1 machine(s)...
Waiting on model to be removed...
Waiting on model to be removed...
Waiting on model to be removed...
Waiting on model to be removed...
Build step 'Execute shell' marked build as failure
Finished: FAILURE
```

check_bundle_cwr_ci:

```
Started by timer
Building in workspace /var/lib/jenkins/jobs/check-bundle-cwr-ci/workspace
[workspace] $ /bin/bash /tmp/hudson7250194468011507768.sh
Running git clone https://github.com/juju-solutions/bundle-cwr-ci --branch bundle-building /tmp/tmp_oxudlrl/
Cloning into '/tmp/tmp_oxudlrl'...
Checking https://github.com/juju-solutions/bundle-cwr-ci
Charms in bunlde ['cs:~kos.tsakalozos/jenkins', 'cs:~juju-solutions/cwr']
Charm jenkins not marked for upgrade.
Upgrading charm cs:~juju-solutions/cwr
Upgrading info {'release': True, 'to-channel': 'beta', 'from-channel': 'edge'}
Running charm show cs:~juju-solutions/cwr -c edge id
id:
  Id: cs:~juju-solutions/cwr-26
  Name: cwr
  Revision: 26
  User: juju-solutions
Latest revision cs:~juju-solutions/cwr-26 in channel edge
Running charm show cs:~juju-solutions/cwr -c edge id
id:
  Id: cs:~juju-solutions/cwr-26
  Name: cwr
  Revision: 26
  User: juju-solutions
Upgraded
Build step 'Execute shell' marked build as failure
Warning: you have no plugins providing access control for builds, so falling back to legacy behavior of permitting any downstream builds to be triggered
Finished: FAILURE
```

File: README.md (hunk 2 above)
Comment (ktsakalozos) [inline README.md]: Done

File: README.md (hunk 1 above)
Comment (ktsakalozos) [inline README.md]: Done

**Pull request merged**